enhancements/alert-on-operator-updates.md

```diff
@@ -0,0 +1,158 @@
+
+- Not a lot to offer in terms of flexibility as compared to OPA (discussed next)
+ 
+### 4. OPA Gatekeeper
+
+##### Glossary 
+- Open Policy Agent ([OPA](https://www.openpolicyagent.org/docs/latest/)) is a policy engine for Cloud Native environments hosted by CNCF.
+- [Gatekeeper](https://github.com/open-policy-agent/gatekeeper) was created to enable users to **_customize_** admission control
+  via configuration, not code and to bring awareness of the cluster’s state, not just the single object under evaluation 
+  at admission time. It enforces policies executed by the Open Policy Agent (OPA).
+  
+#### What is it?
+- We know that an [admission controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/) 
+can be thought of as a gatekeeper that intercept (authenticated) API requests and may change the request object or deny the request altogether.
+OPA Gatekeeper can be thought of as an added layer that allows users to define a custom policy (called `Constraint Template`) which 
+the PDB gets validated against before it is applied on to the cluster.
+
+
+#### How it works?
+- The Gatekeeper project uses a admission webhook to process the admission request whenever a resource in the 
+  cluster is created, updated, or deleted.
+- During the validation process, Gatekeeper acts as a bridge between the API server and OPA. The API server will enforce all policies executed by OPA
+
+#### Steps: 
+1. Install a recent release of [Gatekeeper](https://github.com/open-policy-agent/gatekeeper/blob/master/deploy/gatekeeper.yaml)
+2. Gatekeeper defines two types of Kubernetes custom resources for creating policies: 
+    - `ConstraintTemplates` are templates of an OPA policy and define the parameters needed to consume the template (similar to CRD)
+    - `Constraint` contains the required parameters and what types of Kubernetes events will trigger policy evaluation.  
+3. Once a Constraint template has been deployed in the cluster, an admin can now create individual Constraint CRDs as defined by the Constraint template.
+
+For invalid PDB, for example, you could ideally define that the `minAvailable` is b/w a specific range or that it cannot be set to 0 etc.  
+
+#### Pros
+
+- Offers validating & mutating capabilities for objects that are being applied as under the hood they use admission webhook
+- `Constraint Templates` can be extended to validate different k8 objects. For example, you could say every resource in k8s
+cluster must have an annotation, say email of the owner of the resource 
+- Offers Audit functionality: which allows to detect pre-existing misconfigurations in the cluster
+
+#### Cons
+- Overloading the k8 cluster by installing Gatekeeper
+- Audit functionality has an overhead of Data replication. See [here](https://kubernetes.io/blog/2019/08/06/opa-gatekeeper-policy-and-governance-for-kubernetes/) for details.
+
+
+## 2. Prevent operators from modifying existing SCCs in the cluster
+
+### Design proposal
```

Comment:
OPA was the only design strategy I could think of. I am sure there are more ways do to this. Would  be really helpful, if you could list those down, so that I can read up more on it and update this draft